aten/src/ATen/native/native_functions.yaml

```diff
@@ -6625,6 +6625,24 @@
     CPU: legacy::cpu::_th_histc
     CUDA: _histc_cuda
 
+- func: histogram.bins_tensor_out(Tensor self, Tensor bins, Scalar min=0, Scalar max=0, Tensor? weight=None, bool density=False, *, Tensor(a!) hist, Tensor(b!) bin_edges) -> (Tensor(a!) hist, Tensor(b!) bin_edges)
```

Comment:
One additional TODO: try to make this a structured kernel (we can follow-up with docs + a reference for this)

Reply:
Note: output size of auto bins histogram cannot be supported under structured, as it is data dependent.  (auto bins is not supported in this PR)